engines/sci/graphics/controls32.cpp

```diff
@@ -353,4 +366,303 @@ void GfxControls32::flashCursor(TextEditor &editor) {
 		_nextCursorFlashTick = g_sci->getTickCount() + 30;
 	}
 }
+
+
+reg_t GfxControls32::registerScrollWindow(ScrollWindow *scrollWindow) {
+	_lastScrollWindowId += 1;
+	_scrollWindows[_lastScrollWindowId.getOffset()] = scrollWindow;
+	return _lastScrollWindowId;
+}
+
+ScrollWindow *GfxControls32::getScrollWindow(reg_t id) {
+	Common::HashMap<int, ScrollWindow *>::iterator i;
+	i = _scrollWindows.find(id.getOffset());
+	if (i != _scrollWindows.end())
+		return i->_value;
+	else
+		return nullptr;
+}
+
+void GfxControls32::deregisterScrollWindow(reg_t id) {
+	_scrollWindows.erase(id.getOffset());
+}
+
+
+
+ScrollWindow::ScrollWindow(SegManager *segMan, const Common::Rect &rect,
+                           const Common::Point &point, reg_t plane, uint8 fore,
+                           uint8 back, GuiResourceId font, TextAlign align,
+                           uint8 border)
+: _plane(plane), _foreColor(fore), _backColor(back),
+  _borderColor(border), _fontId(font), _alignment(align), _rect50(rect),
+  _point74(point) {
+
+	_gfxText32 = new GfxText32(segMan, g_sci->_gfxCache);
```

Comment:
This doesn’t need to be separately allocated does it? It could just be a member of the ScrollWindow?